### PR TITLE
Add environment flag for sparse library

### PR DESCRIPTION
**Pull request by DragaDoncila**
In order to support sparse labels passively, we need to set this environment flag to allow `np.asarray` calls on sparse arrays to proceed without raising an error. See [here](https://sparse.pydata.org/en/0.2.0/user_manual/operations/basic.html#auto-densification).

Was really unsure where to put this - we need it before any `add_...` calls are made. Very happy to take suggestions on where this should go.

### CODE REVIEW

**Review by sofroniewn — APPROVED** [review]
I'm ok with this, curious what others think

File: napari/__init__.py
Comment (jni) [inline napari/__init__.py]:
```suggestion
os.environ.setdefault('SPARSE_AUTO_DENSIFY', '1')
```

This will only set it if it is not already set. I'm ok with us adding this setting in general, I'm less ok with overriding someone's explicit setting of 0!

Other than that I'm ok with having this in the init.

**Comment by DragaDoncila** [comment]
I thought we were closer to merging the last PR into sparse than we actually are - we may want to hold off on merging this until that PR is merged.

**Review by jni — APPROVED** [review]

**Comment by jni** [comment]
@DragaDoncila 

> we may want to hold off on merging this until that PR is merged.

nope! =P Best to have everything ready on our end I reckon. =)

**Pull request merged**
